daemon/src/maker_cfd.rs

```diff
@@ -298,66 +310,36 @@ impl Actor {
         self.cfd_feed_actor_inbox
             .send(load_all_cfds(&mut conn).await?)?;
 
-        // Start contract setup
-        tracing::info!("Starting contract setup");
-
-        // Kick-off the CFD protocol
-        let (sk, pk) = crate::keypair::new(&mut rand::thread_rng());
-
-        let margin = cfd.margin()?;
-
-        let maker_params = self.wallet.build_party_params(margin, pk).await?;
-
-        let (actor, inbox) = setup_contract_actor::new(
-            {
-                let inbox = self.takers.clone();
-                move |msg| {
-                    tokio::spawn(inbox.do_send_async(maker_inc_connections::TakerMessage {
-                        taker_id,
-                        command: TakerCommand::OutProtocolMsg { setup_msg: msg },
-                    }));
-                }
-            },
-            setup_contract_actor::OwnParams::Maker(maker_params),
-            sk,
+        let contract_future = setup_contract::new(
+            self.takers.clone().into_sink().with(move |msg| {
+                future::ok(maker_inc_connections::TakerMessage {
+                    taker_id,
+                    command: TakerCommand::Protocol(msg),
+                })
+            }),
+            receiver,
             self.oracle_pk,
             cfd,
             self.wallet.clone(),
+            setup_contract::Role::Maker,
         );
 
-        self.current_contract_setup.replace(inbox.clone());
-
-        for msg in self.contract_setup_message_buffer.drain(..) {
-            inbox.send(msg)?;
-        }
-
-        // TODO: Should we do this here or already earlier or after the spawn?
-        insert_new_cfd_state_by_order_id(
-            order_id,
-            CfdState::ContractSetup {
-                common: CfdStateCommon {
-                    transition_timestamp: SystemTime::now(),
-                },
-            },
-            &mut conn,
-        )
-        .await?;
-        self.cfd_feed_actor_inbox
-            .send(load_all_cfds(&mut conn).await?)?;
-
-        let address = ctx
+        let this = ctx
             .address()
             .expect("actor to be able to give address to itself");
 
         tokio::spawn(async move {
-            address
-                .do_send_async(CfdSetupCompleted {
-                    order_id,
-                    dlc: actor.await,
-                })
+            let dlc = contract_future.await;
+
+            this.do_send_async(CfdSetupCompleted { order_id, dlc })
```

Comment:
❓ Is this one OK to be async or should it also be `send`? 
I think we want to ensure that we handle the completion, but it is not critical to be async. Should be fine being async, but I'm not 100% about side effects.

Reply:
Should be fine because nothing happens after this function, i.e. the current task just dies :)